### 2.1. Problem Statement: STATISTICS 1

In [1]:
import numpy as np

#### 1. You survey households in your area to find the average rent they are paying. Find
#### the standard deviation from the following data:
$1550, $1700, $900, $850, $1000, $950.

In [2]:
rendArray = np.array([1550, 1700,  900, 850,  1000, 950])
avgRent = rendArray.sum()/len(rendArray)
avgRent

1158.3333333333333

In [3]:
rendArray.mean()

1158.3333333333333

#### 2. Find the variance for the following set of data representing trees in California
#### (heights in feet):
#### 3, 21, 98, 203, 17, 9

In [4]:
treeHeights = np.array([3, 21, 98, 203, 17, 9])
#calulate mean
avgHeight = treeHeights.mean()
eleMinusMeanSqr = [(x- avgHeight)**2 for x in treeHeights]
variance = np.sum(eleMinusMeanSqr)/len(treeHeights)
variance

5183.25

In [5]:
np.var(treeHeights)

5183.25

#### 3. In a class on 100 students, 80 students passed in all subjects, 10 failed in one
##### subject, 7 failed in two subjects and 3 failed in three subjects. Find the
##### probability distribution of the variable for number of subjects a student from
##### the given class has failed in.

In [6]:
import matplotlib.pyplot as plt
print('probability of failing in 0 subjects p(X=0)',80/100)
print('probability of failing in 1 subjects p(X=1)',10/100)
print('probability of failing in 2 subjects p(X=2)',7/100)
print('probability of failing in 3 subjects p(X=3)',3/100)

probDis = (80/100,10/100,7/100,3/100)
prob = (1,2,3,4)
plt.hist(probDis)


probability of failing in 0 subjects p(X=0) 0.8
probability of failing in 1 subjects p(X=1) 0.1
probability of failing in 2 subjects p(X=2) 0.07
probability of failing in 3 subjects p(X=3) 0.03


(array([3., 0., 0., 0., 0., 0., 0., 0., 0., 1.]),
 array([0.03 , 0.107, 0.184, 0.261, 0.338, 0.415, 0.492, 0.569, 0.646,
        0.723, 0.8  ]),
 <a list of 10 Patch objects>)

### 2.2. Problem Statement: STATISTICS 2

#### 1. A test is conducted which is consisting of 20 MCQs (multiple choices questions)
#### with every MCQ having its four options out of which only one is correct.
#### Determine the probability that a person undertaking that test has answered
#### exactly 5 questions wrong.

In [7]:
from scipy.stats import binom

In [8]:
n= 20 
r = 5
#the probability of failuer = probability of giving a right answer
s = 1/4
#the probability of success = probability of giving a wrong answer (1 - s) 
q = 3/4
# Its a binomial distribution
# p(X = 5) = ncr * q**r * s**n-r

import scipy.special as sp
ncr = sp.comb(20,5)
probOf5QuesWrong = ncr * q**r * s**(n-r)
probOf5QuesWrong

3.4264958230778575e-06

In [9]:
binom.pmf(5, 20, 3/4)

3.4264958230778435e-06

#### 2. A die marked A to E is rolled 50 times. Find the probability of getting a
#### “D” exactly 5 times.

In [10]:
n =50
r= 5
#the probability of gettinng "D" success
q = 1/5

#the probability of  not getting D failuer
s = 1 - q
ncr = sp.comb(50,5)
probOfDGetting5Times = ncr * q**r * s**(n-r)
probOfDGetting5Times

0.029531204310524296

In [11]:
binom.pmf(5, 50, 1/5)

0.029531204310523224

#### 3. Two balls are drawn at random in succession without replacement from an urn
#### containing 4 red balls and 6 black balls.
#### Find the probabilities of all the possible outcomes.

In [12]:
redballs = 4
blackballs = 6
totalballs = redballs+blackballs
#probabilty of red balls  redballs/totalballs = 4/10
#probabilty of black balls  blackballs/totalballs = 6/10
#draw two balls without replacement the above is example of conditional probability
#The possible outcomes (R,R),(R,B),(B,R),(B,B)
#P((R,R)) = P(R) * P(R/R) = (4/10) * (3/9) = 2/15

#P((R,B)) = P(R) * P(R/B) = (4/10) * (6/9) = 4/15

#P((B,R)) = P(B) * P(B/R) = (6/10) * (4/9) = 4/15

#P((B,B)) = P(B) * P(B/B) = (6/10) * (5/9) = 1/3

print("The probability of (R,R) is",round((4/10)*(3/9),2))
print("The probability of (R,B) is",round((4/10)*(6/9),2))
print("The probability of (B,R) is",round((6/10)*(4/9),2))
print("The probability of (B,B) is",round((6/10)*(5/9),2))

The probability of (R,R) is 0.13
The probability of (R,B) is 0.27
The probability of (B,R) is 0.27
The probability of (B,B) is 0.33


### 2.3 Problem Statement: STATISTICS 3 

Problem Statement 1:
Blood glucose levels for obese patients have a mean of 100 with a standard deviation
of 15. A researcher thinks that a diet high in raw cornstarch will have a positive
effect on blood glucose levels. A sample of 36 patients who have tried the raw
cornstarch diet have a mean glucose level of 108. Test the hypothesis that the raw
cornstarch had an effect or not.

In [13]:
import math
population_mean = 100
population_sigma = 15
samplesize = 36
sample_mean = 108

#step1: State the hypothesis. The polpulation mean 100
# H0: μ=100  >>>> null hypothesis
# H0: μ≠100  >>>> alternate hypothesis or research hypothesis

#step2: set up significance level. If it is not given in the problem let us assume 5%(0.05)
#This 5% called as significance level also knnow as alpha level(symbolized as α).

#(1- Significance level) is also known as Confidence Level 
#i.e. we can say that I am 95% confident that it is not driven by randomness.

#step3: Claculate Z Score
Z = (sample_mean - population_mean)/(population_sigma/math.sqrt(samplesize))
print("Z Score: ",Z)

#step: look at the Z ScoreTable get the p value associated  with 3.20
#Associated p value is 0.9993
#The probability of having glucose level less than 108 is 0.9993 and more than or equal to 108 (1 - 0.9993)=0.0007

#Step-4: Sice the probability of having mean glucose level more than or equals to 108  is 0.0007 which is  less than 0.05 
#so we will reject the Null hypothesis i.e. there is raw cornstarch effect.

Z Score:  3.2


Problem Statement 2:
In one state, 52% of the voters are Republicans, and 48% are Democrats. In a
second state, 47% of the voters are Republicans, and 53% are Democrats.
Suppose a simple random sample of 100 voters are surveyed from each state.

What is the probability that the survey will show a greater percentage of
Republican voters in the second state than in the first state?

In [14]:
#let :-
#P1 = the proportion of Republican voters in the first state
#P2 = the proportion of Republican voters in the second state


#The number of voters sampled from the first state (n1) = 100
n1 = 100
#The number of voters sampled from the second state (n2) = 100.
n2 = 100

P1 = 0.52
#(1 - P1) = Q1
Q1 = 0.48
P2 = 0.47
#(1 - P2) = Q2
Q2 = 0.53


#The mean of the difference in sample proportions .i.e Expected Value E[P1 -P2] = P1 - P2 = mu
mu =  P1 - P2

#The standard deviation of the difference (std)
std = math.sqrt(((P1 * Q1 ) / n1) + ((P2 * Q2) /n2))
print("Mu : ",mu,"Std : ",std)

#This problem requires us to find the probability that P1 is less than P2. 
#This is equivalent to finding the probability that P1 - P2 < 0. 
x = 0
#To find this probability, we need to transform the random variable (P1 - P2) into a z-score. 
#That transformation appears below.
Z_p1_p2 = ( x - mu)/std

print("Z_score(p1,p2) : ",Z_p1_p2)

#From Z table we find that the probability of a z-score being -0.7082 or less is 0.24.

#Therefore, the probability that the survey will show a greater percentage of Republican voters 
#in the second state than in the first state is 0.24.


Mu :  0.050000000000000044 Std :  0.07061869440877536
Z_score(p1,p2) :  -0.7080278164104213


Problem Statement 3
You take the SAT and score 1100. The mean score for the SAT is 1026 and the
standard deviation is 209. How well did you score on the test compared to the
average test taker?

In [15]:
x =1100 #sample mean
mu = 1026 #population mean
std = 209 #population std
#caclulate Z-Score
Z = (x-mu)/std

print("Z Score : ",Z)
#the above calculation shows that my score is 0.35 standard deviations above the mean
print("My Score is in the range {} - {}  with a  zscore {:.2f}".format(mu - std,mu + std,Z))


Z Score :  0.35406698564593303
My Score is in the range 817 - 1235  with a  zscore 0.35


### 2.4 Problem Statement: STATISTICS 4

1. Is gender independent of education level? A random sample of 395 people were
surveyed and each person was asked to report the highest education level they
obtained. The data that resulted from the survey is summarized in the
following table:
     High-School Bachelor Masters Ph.D. Total
Female  60         54        46    41    201
Male    40         44        53    57    194
Total   100        98        99    98    395
Question: Are gender and education level dependent at 5% level of
significance? In other words, given the data collected above, is there a
relationship between the gender of an individual and the level of education
that they have obtained?

##### Chi-Square indipendence
##### H0:Null Hypothesis: Thest gender and eduation catagorical variable are indepndent
##### H1:alternate Hypothesis: Thest gender and eduation catagorical variable are dependent
##### level of significace =5%= 0.05
        

In [18]:
import numpy as np
import pandas as pd
import scipy.stats as stats
female_list = [60,54,46,41]
male_list = [40,44,53,57]

marks = female_list + male_list
marks

sex = ["Female","Female","Female","Female","Male","Male","Male","Male"]
education = ["High-School", "Bachelor", "Masters", "Ph.D.","High-School", "Bachelor", "Masters", "Ph.D."]
df_education = pd.DataFrame({"Sex":sex,"Education":education,"Marks":marks})
print(df_education)

      Sex    Education  Marks
0  Female  High-School     60
1  Female     Bachelor     54
2  Female      Masters     46
3  Female        Ph.D.     41
4    Male  High-School     40
5    Male     Bachelor     44
6    Male      Masters     53
7    Male        Ph.D.     57


In [19]:
df2 = pd.crosstab(df_education.Sex, df_education.Education,df_education.Marks, aggfunc="sum",margins=True)
df2

Education  Bachelor  High-School  Masters  Ph.D.  All
Sex                                                  
Female           54           60       46     41  201
Male             44           40       53     57  194
All              98          100       99     98  395

In [20]:
df2.columns = ["Bachelors","High School","Masters","Ph.d.","row_totals"]

df2.index = ["Female","Male","col_totals"]
df2

Bachelors  High School  Masters  Ph.d.  row_totals
Female             54           60       46     41         201
Male               44           40       53     57         194
col_totals         98          100       99     98         395

In [26]:
#Excluding total fields
observed = df2.iloc[0:2,0:4]
observed

Bachelors  High School  Masters  Ph.d.
Female         54           60       46     41
Male           44           40       53     57

In [31]:
expedted = np.outer(df2["row_totals"][0:2],df2.loc["col_totals"][0:4])/395.0
expedted

array([[49.86835443, 50.88607595, 50.37721519, 49.86835443],
       [48.13164557, 49.11392405, 48.62278481, 48.13164557]])

In [37]:
chi_squared_stat = ((observed - expedted)**2/expedted).sum().sum()
chi_squared_stat

8.006066246262538

In [40]:
stats.chi2_contingency(observed= observed)

(8.006066246262538,
 0.045886500891747214,
 3,
 array([[49.86835443, 50.88607595, 50.37721519, 49.86835443],
        [48.13164557, 49.11392405, 48.62278481, 48.13164557]]))

Note: The degrees of freedom for a test of independence equals the product of the number of categories in each variable minus 1. In this case we have a 2x4 table so df = 1x3 = 3.

Check chi square table critical value where degree of freedome = 3 and significance level 5% = 0.05
The critical value with 3 degree of freedom is 7.815. 8.006> 7.815.
therefore we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance.


In [ ]:
2.Using the following data, perform a one-way analysis of variance using α=.05.
Write up the results in APA format.
[Group1: 51, 45, 33, 45, 67]
[Group2: 23, 43, 23, 43, 45]
[Group3: 56, 76, 74, 87, 56]

In [41]:
#The analysis of variance or ANOVA is a statistical inference test that lets you compare multiple groups at the same time.
#The one-way ANOVA tests whether the mean of some numeric variable differs across the levels of one categorical variable. 
#It essentially answers the question: do any of the group means differ from one another? 

#The scipy library has a function for carrying out one-way ANOVA tests called scipy.stats.f_oneway()
import scipy.stats as stats
Group1 = [51, 45, 33, 45, 67]
Group2 = [23, 43, 23, 43, 45]
Group3 = [56, 76, 74, 87, 56]
# Perform the ANOVA
statistic, pvalue = stats.f_oneway(Group1,Group2,Group3)
print("F Statistic value {} , p-value {}".format(statistic,pvalue))
if pvalue < 0.05:
    print('True')
else:
    print('False')

F Statistic value 9.747205503009463 , p-value 0.0030597541434430556
True


Note The test result suggests the groups don't have the same sample means in this case, since the p-value is significant at a 99% confidence level. Here the p-value returned is 0.00305 which is < 0.05

In [ ]:
3.Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25.
For 10, 20, 30, 40, 50:

In [42]:
stats.f_oneway([10, 20, 30, 40, 50],[5,10,15, 20, 25])

F_onewayResult(statistic=3.6, pvalue=0.0943497728424377)

In [43]:
Group1 = [10, 20, 30, 40, 50]
Group2 = [5,10,15, 20, 25]

mean_1 = np.mean(Group1)
mean_2 = np.mean(Group2)

grp1_sub_mean1 = []
grp2_sub_mean2 = []
add1 = 0
add2 = 0
for items in Group1:
    add1 += (items - mean_1)**2
for items in Group2:
    add2 += (items - mean_2)**2
var1 = add1/(len(Group1)-1)
var2 = add2/(len(Group2)-1)

F_Test = var1/var2
print("F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25 is : ",F_Test)

F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25 is :  4.0
